includes/validation/class-amp-validation-error-taxonomy.php

```diff
@@ -771,6 +778,49 @@ public static function add_error_type_clauses_filter() {
 		}, 10, 2 );
 	}
 
+	/**
+	 * Allows the single error post.php page to filter by error and support searches.
+	 *
+	 * This page is really more like an edit-tags.php page,
+	 * in that it has a WP_Terms_List_Table of the terms associated with the error.
+	 * On filtering this by error type, it sends a POST request to post.php.
+	 * It then redirects the user.
+	 * So this filters the redirect URL to include query vars for the error type and 's', if they are present.
+	 * When the page loads, the errors are filtered by type and/or the search.
+	 *
+	 * @param string $url The redirect URL.
+	 * @param int    $post_id The post ID.
+	 * @return int $url The filtered redirect URL.
+	 */
+	public static function add_to_post_redirect_url( $url, $post_id ) {
```

Comment:
As noted above, all of this needs to be done in JS instead.

Reply:
[These commits]https://github.com/Automattic/amp-wp/compare/c4ed1640644394fe031e105b4c0115cdb9b4f440...a5521682ef24e70f239536859a0e3c9d744e16ac) allow filtering, using JavaScript. I still need to implement the search box with JS.